pennylane_lightning/lightning_kokkos/lightning_kokkos.py

```diff
@@ -69,799 +55,778 @@
+
+    @property
+    def state_vector(self):
+        """Returns a handle to the statevector."""
+        return self._kokkos_state
+
+    def _apply_state_vector(self, state, device_wires):
+        """Initialize the internal state vector in a specified state.
+
+        Args:
+            state (array[complex]): normalized input state of length ``2**len(wires)``
+                or broadcasted state of shape ``(batch_size, 2**len(wires))``
+            device_wires (Wires): wires that get initialized in the state
+        """
+
+        if isinstance(state, self._kokkos_state.__class__):
+            state_data = allocate_aligned_array(state.size, np.dtype(self.C_DTYPE), True)
+            state.DeviceToHost(state_data)
+            state = state_data
 
-            if (
-                isinstance(observable, (qml.ops.Hamiltonian, qml.Hermitian))
-                or (observable.arithmetic_depth > 0)
-                or isinstance(observable.name, List)
+        ravelled_indices, state = self._preprocess_state_vector(state, device_wires)
+
+        # translate to wire labels used by device
+        device_wires = self.map_wires(device_wires)
+        output_shape = [2] * self.num_wires
+
+        if len(device_wires) == self.num_wires and Wires(sorted(device_wires)) == device_wires:
+            # Initialize the entire device state with the input state
+            self.sync_h2d(self._reshape(state, output_shape))
+            return
+
+        self._kokkos_state.setStateVector(ravelled_indices, state)  # this operation on device
+
+    def _apply_basis_state(self, state, wires):
+        """Initialize the state vector in a specified computational basis state.
+
+        Args:
+            state (array[int]): computational basis state of shape ``(wires,)``
+                consisting of 0s and 1s.
+            wires (Wires): wires that the provided computational state should be initialized on
+
+        Note: This function does not support broadcasted inputs yet.
+        """
+        num = self._get_basis_state_index(state, wires)
+        self._create_basis_state(num)
+
+    def _apply_lightning_midmeasure(self, operation: MidMeasureMP, mid_measurements: dict):
+        """Execute a MidMeasureMP operation and return the sample in mid_measurements.
+        Args:
+            operation (~pennylane.operation.Operation): mid-circuit measurement
+        Returns:
+            None
+        """
+        wires = self.wires.indices(operation.wires)
+        wire = list(wires)[0]
+        sample = qml.math.reshape(self.generate_samples(shots=1), (-1,))[wire]
+        if operation.postselect is not None and sample != operation.postselect:
+            mid_measurements[operation] = -1
+            return
+        mid_measurements[operation] = sample
+        getattr(self.state_vector, "collapse")(wire, bool(sample))
+        if operation.reset and bool(sample):
+            self.apply([qml.PauliX(operation.wires)], mid_measurements=mid_measurements)
```

Comment:
Do we need to update test coverage for here?

Reply:
That one is strange because the tests are running and hitting these lines AFAIK.